src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -409,15 +409,25 @@ ValidatorList::for_each_listed (
 }
 
 std::size_t
-ValidatorList::calculateQuorum (std::size_t nTrustedKeys)
+ValidatorList::calculateMinimumQuorum (
+    std::size_t nListedKeys, bool unlistedLocal)
 {
-    // Use 80% for large values of n, but have special cases for small numbers.
-    constexpr std::array<std::size_t, 10> quorum{{ 0, 1, 2, 2, 3, 3, 4, 5, 6, 7 }};
-
-    if (nTrustedKeys < quorum.size())
-        return quorum[nTrustedKeys];
-
-    return nTrustedKeys - nTrustedKeys / 5;
+    if (nListedKeys == 0)
```

Comment:
This make me raise an eyebrow. 

I'd be more inclined to simply remove the `if` and return `1` from the `if` below. I don't think anything would break and I know your tests suggest nothing breaks. @JoelKatz, @bachase?